src/components/cards/RelationshipsCard.jsx

```diff
@@ -1,106 +1,401 @@
+
+  // const theme = useTheme();
+  const [confirmDeleteDialog, setConfirmDeleteDialog] = useState({
+    relationshipGuid: null,
+    open: false,
+  });
+  const [
+    openRelationshipDialog,
+    setOpenRelationshipDialog,
+  ] = useState(false);
+  const [selectedIndividualId, setSelectedIndividualId] = useState(
+    null,
+  );
+  const [
+    selectedIndividualFirstName,
+    setSelectedIndividualFirstName,
+  ] = useState(null);
+  const [currentRoles, setCurrentRoles] = useState(null);
+  const [currentType, setCurrentType] = useState(null);
+  const [currentRole1, setCurrentRole1] = useState(null);
+  const [currentRole2, setCurrentRole2] = useState(null);
+  const allValid =
+    selectedIndividualId &&
+    currentType &&
+    currentRole1 &&
+    currentRole2;
+  const types = useMemo(
+    () => {
+      const possibleRelationships = get(
+        siteSettings,
+        ['relationship_type_roles', 'value'],
+        [],
+      );
+      const _types = Object.values(possibleRelationships);
+      setCurrentRoles(get(_types, 'roles', []));
+      return _types;
+    },
+    [siteSettings],
+  );
+  const relationshipTableData = useMemo(
+    () =>
+      map(
+        relationships,
+        relationship => {
+          const selfIndividualMember = find(
+            get(relationship, 'individual_members'),
+            individualMember =>
+              individualMember.individual_guid === individualGuid,
+          );
+          const nonSelfIndividualMember = find(
+            get(relationship, 'individual_members'),
+            individualMember =>
+              individualMember.individual_guid !== individualGuid,
+          );
+          return {
+            guid: relationship?.guid,
+            nonSelfFirstName:
+              nonSelfIndividualMember?.individual_first_name,
+            nonSelfGuid: nonSelfIndividualMember?.individual_guid,
+            type: relationship?.type_label,
+            role: selfIndividualMember?.individual_role_label,
+          };
+        },
+        [],
+      ),
+    [relationships],
+  );
+
+  const onDelete = async relationshipGuid => {
+    const response = await deleteRelationship({
+      relationshipGuid,
+      individualGuid,
+    });
+    if (response?.status === 204)
+      setConfirmDeleteDialog({
+        realtionshipGuid: null,
+        open: false,
+      });
+  };
+
+  const onChangeType = newType => {
+    setCurrentType(newType);
+    setCurrentRole1(null);
+    setCurrentRole2(null);
+    setCurrentRoles(get(newType, 'roles', []));
+  };
+
+  const onChangeRole1 = newRole => {
+    setCurrentRole1(newRole);
+  };
+
+  const onChangeRole2 = newRole => {
+    setCurrentRole2(newRole);
+  };
+
+  const onCloseDialog = () => {
+    setSelectedIndividualId(null);
+    setSelectedIndividualFirstName(null);
+    setOpenRelationshipDialog(false);
+    setCurrentType(null);
+    setCurrentRole1(null);
+    setCurrentRole2(null);
+    clearPostRelationshipError();
+  };
```

Comment:
Do you also want to run `clearDeleteRelationshipError()`?

Reply:
Good catch!